Don't check running container at create time

hqhq opened the pull request:
We should let user create container even if the container he wants
join is not running, that check should be done at start time.

In this case, the running check is done by getIpcContainer() when
we start container.

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

runcom commented:
isn't this something related to having linked containers sharing the namespace (net) even if they are not running? If we go with this, then ipc, net and other namespaces should not require the container running but this is something bigger than just removing c.IsRunning(), @hqhq am I missing something?

hqhq commented:
Yes, check at creation time. Some users create containers at one time, and start them at another.

cpuguy83 commented:
Is there a check on start? Can you add a test?

hqhq commented:
@cpuguy83 Yes, there is check on start, with getIpcContainer(), test added.

cpuguy83 commented:
@hqhq thanks, can you add a negative test as well? ie, trying to start a container that is using a shared ipc with a container that is not started.

hqhq committed:
Don't check running container at create time

We should let user create container even if the container he wants
join is not running, that check should be done at start time.

In this case, the running check is done by getIpcContainer() when
we start container.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

hqhq commented:
@cpuguy83 Yes I already have that test, but made a mistake before, I used `--net` not `--ipc`, now fixed, thanks.

cpuguy83 commented:
LGTM

LK4D4 commented:
LGTM
